### PR TITLE
Prevent users from accessing courses

### LINKED ISSUE #2644: Students can still access pages for archived courses
We have only removed it from the main page.
So if they edit the url they can still get to the page.

Also, we should make sure that students in the NULL section of a course cannot access the page if the edit the url (even if the course is non-archived.)

### PR DESCRIPTION
**Pull request by ropern123**
Prevents users from accessing courses if they are not an instructor and either are in the null registration section or if the course is archived.

closes #2644

### CODE REVIEW

File: site/app/libraries/Access.php [hunk 1]
```diff
@@ -82,6 +82,8 @@ class Access {
     /** If the current set of flags requires the "gradeable_version" argument */
     const REQUIRE_ARG_VERSION           = 1 << 27;
 
+    /** If the current set of flags requires that the course status be checked */
+    const CHECK_COURSE_STATUS           = 1 << 31;
```
Comment (CouleeApps) [inline site/app/libraries/Access.php]: Move this up to be with the other checks

File: site/app/libraries/Access.php [hunk 2]
```diff
@@ -106,6 +108,8 @@ class Access {
     public function __construct(Core $core) {
         $this->core = $core;
 
+        $this->permissions["view_course"] = self::ALLOW_MIN_STUDENT | self::CHECK_COURSE_STATUS;
```
Comment (CouleeApps) [inline site/app/libraries/Access.php]: Change to `course.view` to be consistent with how the scope stuff works

File: site/app/libraries/Access.php [hunk 3]
```diff
@@ -82,6 +82,8 @@ class Access {
     /** If the current set of flags requires the "gradeable_version" argument */
     const REQUIRE_ARG_VERSION           = 1 << 27;
 
+    /** If the current set of flags requires that the course status be checked */
```
Comment (CouleeApps) [inline site/app/libraries/Access.php]: Should document that this does not apply to instructors

**Comment by ropern123** [comment]
users who are locked out of a course cannot log out on that page

File: site/app/libraries/database/DatabaseQueries.php [hunk 4]
```diff
@@ -2088,6 +2088,11 @@ public function getArchivedCoursesById($user_id, $submitty_path) {
         return $return;
     }
 
+    public function getCourseStatus($semester, $course) {
+        $this->submitty_db->query("SELECT c.status FROM courses c WHERE c.semester=? AND c.course=?", array($semester, $course));
```
Comment (MasterOdin) [inline site/app/libraries/database/DatabaseQueries.php]: No point having aliases if doing a select from a single table.

**Pull request merged**